test/e2e_node/device_plugin_test.go

```diff
@@ -803,13 +803,11 @@ func testDevicePluginNodeReboot(f *framework.Framework, pluginSockDir string) {
 
 		ginkgo.BeforeEach(func(ctx context.Context) {
 			ginkgo.By("Wait for node to be ready")
-			gomega.Eventually(ctx, func(ctx context.Context) error {
-				_, err := e2enode.TotalReady(ctx, f.ClientSet)
-				if err != nil {
-					return err
-				}
-				return nil
-			}, time.Minute, time.Second).Should(gomega.Equal(1), "one node should be ready")
+			gomega.Eventually(ctx, func(ctx context.Context) bool {
+				nodes, err := e2enode.TotalReady(ctx, f.ClientSet)
+				framework.ExpectNoError(err)
+				return nodes == 1
+			}, time.Minute, time.Second).Should(gomega.BeTrueBecause("node should be ready"))
```

Comment:
Don't make assertions inside a `gomega.Eventually`. They interact poorly with each other (test is marked as failed immediately, polling may or may not continue).

Instead, return the count and the error, then use `gomega.Equal(1)`.

Reply:
/cancel lgtm

Reply:
Done.